src/commands.ts

```diff
@@ -37,14 +37,23 @@ export class Commands {
 		this.signer = signer;
 		this.config = config || new ConfigHelper().getConfig(network);
 		this.providerUrl = process.env.PROVIDER_URL || this.config.providerUri;
-		if (this.config.chainId === 8996 && os.type() === "Darwin") {
+		if (
+			!process.env.PROVIDER_URL &&
+			this.config.chainId === 8996 &&
+			os.type() === "Darwin"
+		) {
 			this.macOsProviderUrl = "http://127.0.0.1:8030";
-			this.config.metadataCacheUri = "http://127.0.0.1:5000";
 		}
 		console.log("Using Provider :", this.providerUrl);
 		this.macOsProviderUrl &&
 			console.log(" -> MacOS provider url :", this.macOsProviderUrl);
```

Comment:
We should update these console logs to use our Logger

Reply:
the cli does not have any logger class atm, 
Agree with you that we may need a Logger class but i think we should have a new enhancement issue for it first and handle it another PR instead of changing the original purpose of the PR.